src/container.ts

```diff
@@ -122,24 +122,26 @@ export interface ResolveOptions {
 /**
  * Cache entry.
  */
-export interface CacheEntry {
+export interface CacheEntry<T = any> {
   /**
    * The resolver that resolved the value.
    */
-  resolver: Resolver<any>
+  resolver: Resolver<T>
   /**
    * The resolved value.
    */
-  value: any
+  value: T
 }
 
 /**
  * Register a Registration
  * @interface NameAndRegistrationPair
  */
-export interface NameAndRegistrationPair {
-  [key: string]: Resolver<any>
-}
+export type NameAndRegistrationPair<T> = Partial<
+  {
+    [U in keyof T]: Resolver<T[U]>
```

Comment:
Remove `Partial` and use

```suggestion
    [U in keyof T]?: Resolver<T[U]>
```

Reply:
👍